oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/function/SingleServerApplianceToImage.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jclouds.oneandone.rest.compute.function;
+
+import com.google.common.base.Function;
+import org.apache.jclouds.oneandone.rest.domain.SingleServerAppliance;
+import org.jclouds.compute.domain.Image;
+import org.jclouds.compute.domain.ImageBuilder;
+import org.jclouds.compute.domain.OperatingSystem;
+import org.jclouds.compute.domain.OsFamily;
+
+public class SingleServerApplianceToImage implements Function<SingleServerAppliance, Image> {
+
+   @Override
+   public Image apply(SingleServerAppliance from) {
+      OsFamily osFamily = parseOsFamily(from.osVersion());
+      OperatingSystem os = OperatingSystem.builder()
+              .description(osFamily.value())
+              .family(osFamily)
+              .version(parseVersion(from.os()))
+              .is64Bit(is64Bit(from.osArchitecture()))
+              .build();
+
+      return new ImageBuilder()
+              .ids(from.id())
+              .name(from.name())
+              .status(Image.Status.AVAILABLE)
+              .operatingSystem(os)
+              .build();
+   }
+
+   static String parseVersion(String from) {
+      if (from != null) {
+         String[] split = from.toLowerCase().split("^\\D*(?=\\d)");
+
+         if (split.length >= 2) {
+            return split[1];
+         }
+      }
+      return "non";
+   }
+
+   static OsFamily parseOsFamily(String osFamily) {
+      if (osFamily != null) {
+         if (osFamily.toLowerCase().contains("windows")) {
+            return OsFamily.WINDOWS;
+
+         }
+         if (osFamily.toLowerCase().contains("ubuntu")) {
+            return OsFamily.UBUNTU;
+         }
+
+         if (osFamily.toLowerCase().contains("debian")) {
+            return OsFamily.DEBIAN;
+         }
+
+         if (osFamily.toLowerCase().contains("centos")) {
+            return OsFamily.CENTOS;
+         }
+
+         if (osFamily.toLowerCase().contains("ArchLinux")) {
+            return OsFamily.ARCH;
+         }
+      }
+      return OsFamily.UNRECOGNIZED;
```

Comment:
Better do something like https://github.com/jclouds/jclouds-labs/pull/353 to consider all known OsFamilies.